### PR TITLE
[#93498386] Support individual rating

### PR DESCRIPTION
**Pull request by narusemotoki**
about time, quality and response

### CODE REVIEW

**Comment by hayate** [comment]
LGTM
can you please update the version and create a new tag (release)
https://github.com/gengo/gengo-java/blob/master/src/main/java/com/gengo/client/JsonHttpApi.java#L42

File: src/main/java/com/gengo/client/GengoClient.java [hunk 1]
```diff
@@ -205,10 +214,57 @@ public JSONObject approveTranslationJob(int id, Rating rating,
         }
     }
 
+    /**
+     * Approve a translation.
+     * @param id The job ID
+     * @param rating A rating for the translation
```
Comment (yugui) [inline src/main/java/com/gengo/client/GengoClient.java]: Describe that it rates both quality and response.

BTW, is it really correct behavior?  Should rating_quality and rating_time be null here?

File: src/main/java/com/gengo/client/GengoClient.java [hunk 2]
```diff
@@ -172,14 +172,17 @@ public JSONObject reviseTranslationJob(int id, String comments)
...
+     * @param rating_response A rating for the translator response
      * @param commentsForTranslator Comments for the translator
      * @param commentsForGengo Comments for Gengo
      * @param feedbackIsPublic true iff the feedback can be shared publicly
      * @return the response from the server
      * @throws GengoException
      */
-    public JSONObject approveTranslationJob(int id, Rating rating,
+    public JSONObject approveTranslationJob(int id, Rating rating_time,
+            Rating rating_quality, Rating rating_response,
```
Comment (yugui) [inline src/main/java/com/gengo/client/GengoClient.java]: Use camelCase for local variables and parameters in Java.

File: src/main/java/com/gengo/client/GengoClient.java [hunk 3]
```diff
@@ -205,10 +214,57 @@ public JSONObject approveTranslationJob(int id, Rating rating,
...
+     * @param rating_time A rating for the translation time
+     * @param rating_quality A rating for the translation quality
+     * @param rating_response A rating for the translator response
+     * @param commentsForTranslator Comments for the translator
+     * @param commentsForGengo Comments for Gengo
+     * @return the response from the server
+     * @throws GengoException
+     */
+    public JSONObject approveTranslationJob(int id, Rating rating_time,
+            Rating rating_quality, Rating rating_response,
```
Comment (yugui) [inline src/main/java/com/gengo/client/GengoClient.java]: use camelCase.

**Comment by yugui** [comment]
@randomwire This PR contains a kind of user-facing documentation.  Please review it from the perspective of UX and English writing.

**Comment by randomwire** [comment]
@narusemotoki could you use the following improved param names + descriptions:

@param id 					The job ID
@param rating_time 			Rating of the translation time/speed
@param rating_quality 			Rating of the translation quality
@param rating_response 		Rating of the translator responsiveness
@param feedbackForTranslator 	Feedback for the translator
@param feedbackForGengo 		Feedback for Gengo
@param feedbackIsPublic 		Whether the src/tgt text & feedback can be shared publicly 
@return 						Response from the server
@throws 						GengoException

File: src/main/java/com/gengo/client/GengoClient.java (hunk 3 above)
Comment (narusemotoki) [inline src/main/java/com/gengo/client/GengoClient.java]: Thank you!

**Comment by seikoudoku2000** [comment]
LGTM

**Pull request merged**